grails-app/controllers/com/careerfly/organization/interview/InterviewController.groovy

```diff
@@ -0,0 +1,195 @@
+package com.careerfly.organization.interview
+
+import com.careerfly.geo.Address
+import com.careerfly.geo.City
+import com.careerfly.organization.Company
+import com.careerfly.organization.interview.Interview
+import com.careerfly.organization.interview.InterviewRound
+import com.careerfly.taggable.Tag
+import com.careerfly.user.User
+
+class InterviewController {
+
+    def index() { }
+
+    def jobRequirement() {
+
+        println params
+        session.company = params
+    }
+
+    def save() {
+
+        println params
+
+        Interview interviewInstance = new Interview()
+
+        Set workExperience = params.workExperience.split(";")
+        Set tech = params.technologies.split(";")
+        Set tools = params.tools.split(";")
+        Set skills = params.skills.split(";")
+
+        City cityInstance = new City([name: session.company.address])
+        cityInstance.save()
+
+        Address addressInstance = new Address([city: cityInstance])
+        addressInstance.save()
+
+        Company companyInstance = new Company([name: session.company.name, address: addressInstance])
+        companyInstance.save()
+
+        tech.each {
+            Tag techInstance = Tag.findByName(it)
+            if (!techInstance) {
+                techInstance = new Tag(name: it)
+                techInstance.save()
+            }
+            println techInstance
+            println techInstance.errors
+            interviewInstance.technologies.add(techInstance)
+        }
+
+        tools.each {
+            Tag toolInstance = Tag.findByName(it)
```

Comment:
The code inside the `each` look seems repetitive. This could have been made DRY using private methods or services. @Sagars09 @shubhamj15